pkg/kubelet/dockertools/manager.go

```diff
@@ -280,8 +280,15 @@ func (dm *DockerManager) inspectContainer(dockerID, containerName, tPath string)
 		result.status.State.Running = &api.ContainerStateRunning{
 			StartedAt: util.NewTime(inspectResult.State.StartedAt),
 		}
-		if containerName == PodInfraContainerName && inspectResult.NetworkSettings != nil {
-			result.ip = inspectResult.NetworkSettings.IPAddress
+		if dm.networkPlugin.Name() != network.DefaultPluginName && containerName == PodInfraContainerName {
+			netStatus, err := dm.networkPlugin.Status(pod.Namespace, pod.Name, kubeletTypes.DockerID(dockerID))
+			if err == nil {
+				result.ip = netStatus.IPv4.String()
```

Comment:
`NetworkStatus` can be a rich and growing set of useful information. Perhaps keeping it as structured data that is backward compatible would be the right way to go. However consider following scenarios, if we want to change anything to the `NetworkStatus`:
- If a pod is allocated only a v6 address, the above code/logic wouldn't work, i.e. we may need to specify if v4 and/or v6 address is valid
- A pod could be assigned multiple interfaces, and thus multiple IP addresses, juxtaposing all that as `result.ip` may work but I am not sure if it is desired
- If driver wants to display other information like policy, etc.